iree/compiler/Conversion/LinalgToVector/VectorizeLinalgOps.cpp

```diff
@@ -0,0 +1,339 @@
+    if (llvm::all_of(args,
+                     [](Value arg) { return arg.getType().isIntOrFloat(); })) {
+      return failure();
+    }
+    SmallVector<Value, 4> vecArgs;
+    for (Value arg : args) {
+      Value val = legalizeToVectorType(rewriter, arg);
+      if (!val) return failure();
+      vecArgs.push_back(val);
+    }
+    auto vecType = getVecType(rewriter, op.getResult().getType());
+    if (!vecType) return failure();
+    auto newOp = rewriter.create<OpTy>(op.getLoc(), vecType, vecArgs);
+    rewriter.replaceOp(op, newOp.getOperation()->getResults());
+    return success();
+  }
+};
+
+template <typename OpTy>
+struct VectorizeCmpOp : public OpConversionPattern<OpTy> {
+  using OpConversionPattern<OpTy>::OpConversionPattern;
+  LogicalResult matchAndRewrite(
+      OpTy op, ArrayRef<Value> args,
+      ConversionPatternRewriter &rewriter) const override {
+    if (llvm::any_of(args,
+                     [](Value arg) { return arg.getType().isIntOrFloat(); })) {
+      return failure();
+    }
+    auto newOp =
+        rewriter.create<OpTy>(op.getLoc(), op.predicate(), args[0], args[1]);
+    rewriter.replaceOp(op, newOp.getResult());
+    return success();
+  }
+};
+
+struct VectorizeSelectOp : public OpConversionPattern<mlir::SelectOp> {
+  using OpConversionPattern<mlir::SelectOp>::OpConversionPattern;
+  LogicalResult matchAndRewrite(
+      mlir::SelectOp op, ArrayRef<Value> args,
+      ConversionPatternRewriter &rewriter) const override {
+    if (llvm::any_of(args,
+                     [](Value arg) { return arg.getType().isIntOrFloat(); })) {
+      return failure();
+    }
+    auto newOp =
+        rewriter.create<SelectOp>(op.getLoc(), args[0], args[1], args[2]);
+    rewriter.replaceOp(op, newOp.getResult());
+    return success();
+  }
+};
+
+struct VectorizeYieldOp : public OpConversionPattern<linalg::YieldOp> {
+  using OpConversionPattern<linalg::YieldOp>::OpConversionPattern;
+  LogicalResult matchAndRewrite(
+      linalg::YieldOp yieldOp, ArrayRef<Value> args,
+      ConversionPatternRewriter &rewriter) const override {
+    if (!llvm::all_of(args,
+                      [](Value v) { return v.getType().isa<VectorType>(); })) {
+      return failure();
+    }
+    SmallVector<Value, 4> vecArgs;
+    for (Value arg : args) {
+      Value val = legalizeToVectorType(rewriter, arg);
+      if (!val) return failure();
+      vecArgs.push_back(val);
+    }
+    rewriter.replaceOpWithNewOp<linalg::YieldOp>(yieldOp, vecArgs);
+    return success();
+  }
+};
+
+struct VectorizeGenericOp : public OpConversionPattern<linalg::GenericOp> {
+  using OpConversionPattern<linalg::GenericOp>::OpConversionPattern;
+  LogicalResult matchAndRewrite(
+      linalg::GenericOp genericOp, ArrayRef<Value> args,
+      ConversionPatternRewriter &rewriter) const override {
+    // Ignore a case that either a nop pattern or a linalg.fill pattern.
+    if (isa<linalg::YieldOp>(
```

Comment:
I dont think this is needed. It should just pass through fine.

Reply:
This is a tricky one that I found in a model. Through several passes (fusion, canonicalize, etc) we would get a snippet in the dispatch function.

```
%cst = constant 1.0 : f32
linalg.generic { ... } %output_buf
^bb(...):
  linalg.yield %cst
```

There are two other options:

1) Have another pass to rewrite Linalg ops to named ops.
2) Handle this in yield op pattern.

The first approach looks better to me. This is just a workaround to make us be able to run a E2E model.

Reply:
mmm, this looks like broadcast getting into its own dispatch region, it should disappear after this https://github.com/google/iree/commit/15c6d4565b1590ed2fcccb1056db7f7478b95101, no ?

Reply:
oh yes, it's gone. I observed it in MobileNetV2 and it's working without this now. thanks!